File: 4월/230427_pg_두 큐 합 같게 만들기_yyw.java
import java.util.*;
class Solution {
    public int solution(int[] queue1, int[] queue2) {
        int max_length = queue1.length * 3;
    
        Queue<Integer> queue_1 = new LinkedList<>();
        for (int i : queue1){
            queue_1.offer(i);
        }
        Queue<Integer> queue_2 = new LinkedList<>();
        for (int j : queue2){
            queue_2.offer(j);
        }
        
        long queue1_sum = sum(queue1);
        long queue2_sum = sum(queue2);
        
        int cnt = 0;
        while (cnt <= max_length){
            if (queue1_sum > queue2_sum){
                int tmp = queue_1.poll();
                queue_2.offer(tmp);
                queue1_sum -= tmp;
                queue2_sum += tmp;
                cnt++;
            } else if (queue1_sum < queue2_sum){
                int tmp = queue_2.poll();
                queue_1.offer(tmp);
                queue2_sum -= tmp;
                queue1_sum += tmp;
                cnt++;
            }
            if (queue1_sum == queue2_sum){
                return cnt;
            }
        }
        return -1;
    }
    
    public int sum(int[] arr){
        int sumValue = 0;
        for (int a : arr){
            sumValue += a;
        }
        return sumValue;
    }
}

File: 4월/230427_pg_두 큐 합 같게 만들기_yyw.js
function solution(queue1, queue2) {
    let max_cnt = queue1.length * 3;
    let queue1_sum = queue1.reduce((sum, currValue) => {
    return sum + currValue;
    }, 0);
    let queue2_sum = queue2.reduce((sum, currValue) => {
    return sum + currValue;
    }, 0);
    
    let cnt = 0;
    let tmp_elm;
    let [a_idx, b_idx] = [0, 0];
    while (cnt <= max_cnt){
        if (queue1_sum > queue2_sum){
            tmp_elm = queue1[a_idx];
            a_idx++;
            queue2.push(tmp_elm);
            queue1_sum -= tmp_elm;
            queue2_sum += tmp_elm;
            cnt++;
        } else if (queue1_sum < queue2_sum){
            tmp_elm = queue2[b_idx];
            b_idx++;
            queue1.push(tmp_elm);
            queue2_sum -= tmp_elm;
            queue1_sum += tmp_elm;
            cnt++;
        }
        if (queue1_sum == queue2_sum) return cnt;
    }
    return -1;
}